### PR TITLE
feature: SOAP2DAY presence

### LINKED ISSUE #2327: Soap2Day | soap2day.to
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** aidan!#0001 

**What's the name of the service?** Soap2Day | https://soap2day.to

**What should the Presence display?** Something like Netflix or Hulu so others can see what you are watching. 

**If possible, please provide a logo for the service (512x512 minimum)**

### PR DESCRIPTION
**Pull request by fuwn**
Closes #2327.

### CODE REVIEW

File: websites/S/SOAP2DAY/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "SOAP2DAY",
+  "author": {
+    "name": "fun",
+    "id": "217348698294714370"
+  },
+  "category": "socials",
+  "logo": "https://www.google.com/s2/favicons?domain=soap2day.to",
+  "thumbnail": "https://strelizia.cc/zcC8wBhzhFn1aRS5qiLylH7TSy9Lmpiq.png",
```
Comment (ririxidev) [inline websites/S/SOAP2DAY/dist/metadata.json:10]: Only imgur direct links. Logo min. 512x512

File: websites/S/SOAP2DAY/presence.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  "tv":        "Relaxing to some TV",
+}
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "icon",
+    details: constructAction[getAction()],
+    state: ["movie", "tv", "sport"].includes(getAction()) ? getTitle() : "",
+    startTimestamp: new Date().getTime(),
+  };
```
Comment (ririxidev) [inline websites/S/SOAP2DAY/presence.ts:45]: Maybe something more than this? This site has video players so you can get info from it??

File: websites/S/SOAP2DAY/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.3",
+  "service": "SOAP2DAY",
+  "author": {
+    "name": "fun",
+    "id": "217348698294714370"
+  },
+  "category": "socials",
```
Comment (ririxidev) [inline websites/S/SOAP2DAY/dist/metadata.json:8]: I don't think it's social at all.

File: websites/S/SOAP2DAY/presence.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  "faq":       "Reading the FAQ",
+  "tv":        "Relaxing to some TV",
+}
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "icon",
+    details: constructAction[getAction()],
+    state: ["movie", "tv", "sport"].includes(getAction()) ? getTitle() : "",
+    startTimestamp: new Date().getTime(),
```
Comment (ririxidev) [inline websites/S/SOAP2DAY/presence.ts:44]: Math.floor(Date.now / 1000)

**Review by ririxidev — CHANGES_REQUESTED** [review]
.

File: websites/S/SOAP2DAY/presence.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  "home":      "Checking out the home page",
+  "faq":       "Reading the FAQ",
+  "tv":        "Relaxing to some TV",
+}
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "icon",
+    details: constructAction[getAction()],
+    state: ["movie", "tv", "sport"].includes(getAction()) ? getTitle() : "",
```
Comment (ririxidev) [inline websites/S/SOAP2DAY/presence.ts:43]: Ah didn't notice. Never set state/details whatever to "" nor null nor undefined. Do it in different way.

**Review by ririxidev — CHANGES_REQUESTED** [review]
Found something

File: websites/S/SOAP2DAY/dist/metadata.json (hunk 3 above)
Comment (fuwn) [reply websites/S/SOAP2DAY/dist/metadata.json:8]: This was meant to be "stream" or what have you.

File: websites/S/SOAP2DAY/presence.ts (hunk 2 above)
Comment (fuwn) [reply websites/S/SOAP2DAY/presence.ts:45]: Can you elaborate? Info and state is being grabbed;

**Comment by EncryptedDev** [comment]
Fix Codacy too

File: websites/S/SOAP2DAY/presence.ts (hunk 2 above)
Comment (ririxidev) [reply websites/S/SOAP2DAY/presence.ts]: Like if it is paused or and remaining time etc.

Comment (fuwn) [reply websites/S/SOAP2DAY/presence.ts]: I understand, implementing!

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Look at my review

**Review by EncryptedDev — COMMENTED** [review]
bruh

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix your regex and upload proof of it working afterwards

File: websites/S/SOAP2DAY/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "soap2day.im",
+    "soap2day.ac",
+    "soap2day.se",
+    "s2dfree.to",
+    "s2dfree.is",
+    "s2dfree.in",
+    "s2dfree.nl",
+    "s2dfree.cc",
+    "s2dfree.de"
+  ],
```
[the remark this answers is not in the document]
Comment (ririxidev) [reply websites/S/SOAP2DAY/dist/metadata.json:31]: also delete it after doing regexp.

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @fuwn!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/S/SOAP2DAY/dist/metadata.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (fuwn) [reply websites/S/SOAP2DAY/dist/metadata.json:31]: Reason I kept it is the GitHub action complained the file wasn't in the correct format when taking out URL.

**Review by SlowLife1661 — CHANGES_REQUESTED** [review]
`url` is required just keep 1 url, and regex is still wrong I suppose so. I highly suggest you to test it before you push a changes lol ( I mean aren't you supposed to test it first )

File: websites/S/SOAP2DAY/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    "stream",
+    "soap2day",
+    "movies",
+    "tv",
+    "friends"
+  ],
+  "description": {
+    "en": "You can watch all videos without logging in or registration. Soap2day is free all the time, but there are few ads on videoplayer page."
+  },
+  "regExp": "(soap2day|s2dfree)[.]"
```
Comment (StrikerFRFX) [inline websites/S/SOAP2DAY/dist/metadata.json:23]: Regex is still wrong, I suggest you look into how regex work, reference other presences, and test it [here](https://regex101.com/)

**Review by StrikerFRFX — CHANGES_REQUESTED** [review]
Hi @fuwn, 
I've requested changes below in my review which you should take a look at 😄 
 -  In addition to the Regex issue, it is required to have at least `1` URL added into the Metadata.

Thanks,
**Striker**, PreMiD Reviewer Team.
Contact: me@strikerfrfx.dev, Striker#1337

File: websites/S/SOAP2DAY/dist/metadata.json (hunk 7 above)
Comment (Bas950) [reply websites/S/SOAP2DAY/dist/metadata.json:23]: You also still need at least 1 URL, as fallback for if for some reason the regex fails.

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by ririxidev** [comment]
Closing pull request since no fix for ``regex`` is made when was requested by our reviewer 7 days ago.

**Comment by fuwn** [comment]
@ririxidev, I did some homework on RegEx and I think I've got a formula that works!

`([a-z0-9]+[.])*(soap2day|s2dfree)([.][a-z]+)`



Thoughts?

**Comment by fuwn** [comment]
More so, I've done actual tests using the SOAP2DAY site as well, those all work out too.

**Comment by ririxidev** [comment]
after 2 months lul
yeah it's fine - create new pull request then

**Pull request closed without merging**